src/sentry/auth/superuser.py

```diff
@@ -275,12 +282,41 @@ def set_logged_in(self, user, current_datetime=None):
         request = self.request
         if current_datetime is None:
             current_datetime = timezone.now()
+
+        token = get_random_string(12)
+
+        try:
+            su_access_json = json.loads(request.body)
+
+            if "superuserAccessCategory" in su_access_json:
+
+                su_access_info = SuperuserAccessSerializer(data=su_access_json)
+
+                if not su_access_info.is_valid():
+                    raise serializers.ValidationError(su_access_info.errors)
+
+                logger.info(
+                    "superuser.superuser_access",
+                    extra={
+                        "superuser_session_id": token,
+                        "user_id": request.user.id,
+                        "user_email": request.user.email,
+                        "su_access_category": su_access_info.validated_data[
+                            "superuserAccessCategory"
+                        ],
+                        "reason_for_su": su_access_info.validated_data["superuserReason"],
+                    },
+                )
+        except json.JSONDecodeError:
+            # since there isn't a valid request body, we just want to give them su access for now (this is the case of sso login for now)
+            pass
```

Comment:
I'd suggest extracting this new code block out to a helper method. In addition to keeping `set_logged_in` shorter overall, you could shrink the `try` block as:

```python
    def _...(...):
        try:
            su_access_json = json.loads(request.body)
        except json.JSONDecodeError:
            return

        if "superuserAccessCategory" in su_access_json:
        # rest of method...
```